langchain/src/embeddings/llama_cpp.ts

```diff
@@ -0,0 +1,75 @@
+import { LlamaModel, LlamaContext } from "node-llama-cpp";
+import {
+  LlamaBaseCppInputs,
+  createLlamaModel,
+  createLlamaContext,
+} from "../util/llama_cpp.js";
+import { Embeddings, EmbeddingsParams } from "./base.js";
+
+/**
+ * Note that the modelPath is the only required parameter. For testing you
+ * can set this in the environment variable `LLAMA_PATH`.
+ */
+export interface LlamaCppEmbeddingsParams
+  extends LlamaBaseCppInputs,
+    EmbeddingsParams {}
+
+export class LlamaCppEmbeddings extends Embeddings {
+  _model: LlamaModel;
+
+  _context: LlamaContext;
+
+  constructor(inputs: LlamaCppEmbeddingsParams) {
+    super(inputs);
+    const _inputs = inputs;
+    _inputs.embedding = true;
+
+    this._model = createLlamaModel(_inputs);
+    this._context = createLlamaContext(this._model, _inputs);
+  }
+
+  /**
+   * Generates embeddings for an array of texts.
+   * @param texts - An array of strings to generate embeddings for.
+   * @returns A Promise that resolves to an array of embeddings.
+   */
+  async embedDocuments(texts: string[]): Promise<number[][]> {
+    return new Promise((resolve) => {
```

Comment:
We can also take advantage of concurrent calls and retries if we use `this.caller.call` (standard property on `Embeddings`

https://github.com/langchain-ai/langchainjs/blob/main/langchain/src/embeddings/openai.ts#L246